shared-libs/tombstone-utils/src/tombstone-utils.js

```diff
@@ -0,0 +1,96 @@
+var _ = require('underscore');
+
+var TYPES = ['data_record', 'person', 'clinic', 'district_hospital', 'health_center'],
+    TOMBSTONE_TYPE = 'tombstone',
+    TOMBSTONE_ID_SEPARATOR = '____';
+
+module.exports = function(Promise, DB) {
+  var needsTombstone = function (doc) {
+    return TYPES.indexOf(doc.type) !== -1;
+  };
+
+  var generateTombstoneId = function (id, rev) {
+    return [id, rev, TOMBSTONE_TYPE].join(TOMBSTONE_ID_SEPARATOR);
+  };
+
+  var saveTombstone = function(doc, logger) {
+    var tombstone = _.omit(doc, ['_attachments', 'transitions', '_deleted']);
+    var tombstoneDoc = {
+      _id: generateTombstoneId(doc._id, doc._rev),
+      type: TOMBSTONE_TYPE,
+      tombstone: tombstone
+    };
+
+    logger.info('saving tombstone for ' + doc._id);
+
+    return DB
+      .put(tombstoneDoc)
+      .catch(function(error) {
+        if (error.status === 409) {
+          return;
+        }
+
+        throw error;
+      });
+  };
+
+  var getDoc = function(change) {
+    if (change.doc) {
+      return Promise.resolve(change.doc);
+    }
+
+    return DB.get(change.id, { rev: change.changes[0].rev });
+  };
+
+  var getTombstoneRegex = function() {
+    return new RegExp('(.*)' + TOMBSTONE_ID_SEPARATOR + '(.*)' + TOMBSTONE_ID_SEPARATOR + TOMBSTONE_TYPE);
```

Comment:
Store the regex as a var in the module scope so we don't have to recompile it every time it's used.